.github/workflows/cross-compile.yml

```
@@ -0,0 +1,38 @@
+name: Cross Compile
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+jobs:
+  native-gems:
+    strategy:
+      fail-fast: false
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: extractions/setup-just@v1
+      - name: setup ruby 3.3
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.3
+      - name: build and install tree-sitter
+        run: just setup-ts
+      - name: setup
+        run: just setup-bundler
+      - name: compile
+        run: just compile
+      - name: package native gem # TODO: and test them
+        run: just gem-cross
+      - name: release
+        if: github.event_name != 'pull_request'
```

We should be able to only run this when a new tag is created to reduce the number of drafts that get created:

```suggestion
        if: startsWith(github.ref, 'refs/tags/')
```

Example: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif

---

Amazing. Will add it.

---

Actually, I am having doubts.

We want those drafts because when we publish, we create the tag at the same time, so it's already too late.

So now we have to tag first and then publish ?

---

Oh I didn't realize that we didn't have the tag already

---

I'm gonna keep it as such and will try to revisit later when I have time to refocus on the CI … or if you feel like taking a stab at it, be my guest :)

For the time being I'm not bothered by the drafts.